pinot-broker/src/main/java/org/apache/pinot/broker/routing/HelixExternalViewBasedTimeBoundaryService.java

```diff
@@ -111,25 +112,53 @@ public void updateTimeBoundaryService(ExternalView externalView) {
         continue;
       }
 
-      // Convert segment end time into table time unit
-      // NOTE: for now, time unit in segment ZK metadata should always match table time unit, but in the future we might
-      //       want to always use MILLISECONDS as the time unit in segment ZK metadata
-      maxTimeValue = Math.max(maxTimeValue, tableTimeUnit.convert(segmentEndTime, segmentZKMetadata.getTimeUnit()));
+      long segmentEndTimeMs = segmentZKMetadata.getTimeUnit().toMillis(segmentEndTime);
+      if (latestSegmentZKMetadata == null) {
```

Comment:
`if (latestSegmentZkMetadata == null || segmentEndTimeMs > latestSegmentEndTimeMs)`
works?

Reply:
`if (segmentEndTimeMs < latestSegmentEndTimeMs)` is enough. Fixed